packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/HttpLogObfuscatorTests.kt

```diff
@@ -0,0 +1,354 @@
+                if (it.contains(""""password":"***"""")) {
+                    fail("Password was obfuscated: $it")
+                } else if (it.contains(""""access_token":"***","refresh_token":"***"""")) {
+                    fail("Access/refresh tokens were obfuscated: $it")
+                } else if (it.contains(""""key":"***"""")) {
+                    fail("API key was obfuscated: $it")
+                } else if (it.contains(""""id_token":"***"""")) {
+                    fail("Apple/Google ID tokens were obfuscated: $it")
+                } else if (it.contains(""""accessToken":"***"""")) {
+                    fail("Facebook token was obfuscated: $it")
+                } else if (it.contains(""""authCode":"***"""")) {
+                    fail("Google Auth Code was obfuscated: $it")
+                } else if (it.contains(""""token":"***"""")) {
+                    fail("JWT key was obfuscated: $it")
+                } else if (
+                    it.contains(""""arguments":[***]""") ||
+                    it.contains("BODY START\n***\nBODY END")
+                ) {
+                    fail("Custom function arguments were obfuscated: $it")
+                }
+            }
+        }
+    }
+
+    private enum class Operation {
+        OBFUSCATED_PASSWORD,
+        OBFUSCATED_ACCESS_AND_REFRESH_TOKENS,
+        OBFUSCATED_API_KEY,
+        OBFUSCATED_APPLE_OR_GOOGLE_ID_TOKEN,
+        OBFUSCATED_FACEBOOK,
+        OBFUSCATED_GOOGLE_AUTH_CODE,
+        OBFUSCATED_JWT,
+        OBFUSCATED_CUSTOM_FUNCTION
+    }
+
+    @BeforeTest
+    fun setUp() {
+        channel = Channel(1)
+    }
+
+    private fun initApp(): TestApp {
+        return TestApp(
+            appName = "obfuscator",
+            logLevel = LogLevel.DEBUG,
+            customLogger = ObfuscatorLoggerInspector(channel),
+            initialSetup = { app, service ->
+                initializeDefault(app, service)
+                app.addFunction(TestAppInitializer.FIRST_ARG_FUNCTION)
+                app.addFunction(TestAppInitializer.SUM_FUNCTION)
+                app.addFunction(TestAppInitializer.NULL_FUNCTION)
+            }
+        )
+    }
+
+    @AfterTest
+    fun tearDown() {
+        channel.close()
+
+        if (this::app.isInitialized) {
+            app.close()
+        }
+    }
+
+    @Test
+    fun nullObfuscator() = runBlocking {
+        app = TestApp(
+            appName = "null-obfuscator",
+            logLevel = LogLevel.DEBUG,
+            builder = { it.httpLogObfuscator(null) },
+            customLogger = NullObfuscatorLoggerInspector,
+            initialSetup = { app, service ->
+                initializeDefault(app, service)
+                app.addFunction(TestAppInitializer.FIRST_ARG_FUNCTION)
+                app.addFunction(TestAppInitializer.SUM_FUNCTION)
+                app.addFunction(TestAppInitializer.NULL_FUNCTION)
+            }
+        )
+
+        // Create user and log in
+        user = app.createUserAndLogin()
```

Comment:
The way the test is constructed it is a bit hidden what we actually assert. Why not just collect the messages and assert that the framework actually logged the original request/response messages. You would fail this test if you somehow filtered out all messages due to erroneous log level, etc.